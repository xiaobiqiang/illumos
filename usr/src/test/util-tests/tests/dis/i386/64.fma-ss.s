/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2016 Joyent, Inc.
 */

/*
 * Test ADX related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vfmadd132ss	%xmm0, %xmm1, %xmm2
	vfmadd132ss	(%rax), %xmm1, %xmm2
	vfmadd213ss	%xmm0, %xmm1, %xmm2
	vfmadd213ss	(%rax), %xmm1, %xmm2
	vfmadd231ss	%xmm0, %xmm1, %xmm2
	vfmadd231ss	(%rax), %xmm1, %xmm2
	vfmsub132ss	%xmm0, %xmm1, %xmm2
	vfmsub132ss	(%rax), %xmm1, %xmm2
	vfmsub213ss	%xmm0, %xmm1, %xmm2
	vfmsub213ss	(%rax), %xmm1, %xmm2
	vfmsub231ss	%xmm0, %xmm1, %xmm2
	vfmsub231ss	(%rax), %xmm1, %xmm2
	vfnmadd132ss	%xmm0, %xmm1, %xmm2
	vfnmadd132ss	(%rax), %xmm1, %xmm2
	vfnmadd213ss	%xmm0, %xmm1, %xmm2
	vfnmadd213ss	(%rax), %xmm1, %xmm2
	vfnmadd231ss	%xmm0, %xmm1, %xmm2
	vfnmadd231ss	(%rax), %xmm1, %xmm2
	vfnmsub132ss	%xmm0, %xmm1, %xmm2
	vfnmsub132ss	(%rax), %xmm1, %xmm2
	vfnmsub213ss	%xmm0, %xmm1, %xmm2
	vfnmsub213ss	(%rax), %xmm1, %xmm2
	vfnmsub231ss	%xmm0, %xmm1, %xmm2
	vfnmsub231ss	(%rax), %xmm1, %xmm2
.size libdis_test, [.-libdis_test]
