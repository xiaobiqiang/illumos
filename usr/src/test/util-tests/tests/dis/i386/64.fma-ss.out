    libdis_test:      c4 e2 71 99 d0     vfmadd132ss %xmm0,%xmm1,%xmm2
    libdis_test+0x5:  c4 e2 71 99 10     vfmadd132ss (%rax),%xmm1,%xmm2
    libdis_test+0xa:  c4 e2 71 a9 d0     vfmadd213ss %xmm0,%xmm1,%xmm2
    libdis_test+0xf:  c4 e2 71 a9 10     vfmadd213ss (%rax),%xmm1,%xmm2
    libdis_test+0x14: c4 e2 71 b9 d0     vfmadd231ss %xmm0,%xmm1,%xmm2
    libdis_test+0x19: c4 e2 71 b9 10     vfmadd231ss (%rax),%xmm1,%xmm2
    libdis_test+0x1e: c4 e2 71 9b d0     vfmsub132ss %xmm0,%xmm1,%xmm2
    libdis_test+0x23: c4 e2 71 9b 10     vfmsub132ss (%rax),%xmm1,%xmm2
    libdis_test+0x28: c4 e2 71 ab d0     vfmsub213ss %xmm0,%xmm1,%xmm2
    libdis_test+0x2d: c4 e2 71 ab 10     vfmsub213ss (%rax),%xmm1,%xmm2
    libdis_test+0x32: c4 e2 71 bb d0     vfmsub231ss %xmm0,%xmm1,%xmm2
    libdis_test+0x37: c4 e2 71 bb 10     vfmsub231ss (%rax),%xmm1,%xmm2
    libdis_test+0x3c: c4 e2 71 9d d0     vfnmadd132ss %xmm0,%xmm1,%xmm2
    libdis_test+0x41: c4 e2 71 9d 10     vfnmadd132ss (%rax),%xmm1,%xmm2
    libdis_test+0x46: c4 e2 71 ad d0     vfnmadd213ss %xmm0,%xmm1,%xmm2
    libdis_test+0x4b: c4 e2 71 ad 10     vfnmadd213ss (%rax),%xmm1,%xmm2
    libdis_test+0x50: c4 e2 71 bd d0     vfnmadd231ss %xmm0,%xmm1,%xmm2
    libdis_test+0x55: c4 e2 71 bd 10     vfnmadd231ss (%rax),%xmm1,%xmm2
    libdis_test+0x5a: c4 e2 71 9f d0     vfnmsub132ss %xmm0,%xmm1,%xmm2
    libdis_test+0x5f: c4 e2 71 9f 10     vfnmsub132ss (%rax),%xmm1,%xmm2
    libdis_test+0x64: c4 e2 71 af d0     vfnmsub213ss %xmm0,%xmm1,%xmm2
    libdis_test+0x69: c4 e2 71 af 10     vfnmsub213ss (%rax),%xmm1,%xmm2
    libdis_test+0x6e: c4 e2 71 bf d0     vfnmsub231ss %xmm0,%xmm1,%xmm2
    libdis_test+0x73: c4 e2 71 bf 10     vfnmsub231ss (%rax),%xmm1,%xmm2
